src/Nethermind/Nethermind.JsonRpc/Modules/Admin/PeerInfo.cs

```diff
@@ -1,61 +1,153 @@
-        public string Id { get; }
-        public string Host { get; set; }
-        public int Port { get; set; }
-        public string Address { get; set; }
-        public bool IsBootnode { get; set; }
-        public bool IsTrusted { get; set; }
-        public bool IsStatic { get; set; }
-        public string Enode { get; set; }
-
-        public string ClientType { get; set; }
-        public string EthDetails { get; set; }
-        public string LastSignal { get; set; }
-
-        public bool Inbound { get; set; }
+        private static readonly IReadOnlyList<Capability> EmptyCapabilities = Array.Empty<Capability>();
+
+        public string Enode { get; set; } = string.Empty;
+
+        [JsonConverter(typeof(PublicKeyHashedConverter))]
+        public PublicKey Id { get; set; } = null!;
+
+        public string? Name { get; set; }
+
+        [JsonConverter(typeof(CapabilityConverter))]
+        public IReadOnlyList<Capability> Caps { get; set; } = EmptyCapabilities;
+
+        public string? Enr { get; set; }
+
+        public NetworkInfo Network { get; set; } = new();
+
+        public Dictionary<string, object> Protocols { get; set; } = new();
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public NodeClientType? ClientType { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public string? EthDetails { get; set; }
+
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        public DateTime? LastSignal { get; set; }
 
         public PeerInfo()
         {
         }
 
-        public PeerInfo(Peer peer, bool includeDetails)
+        public PeerInfo(Peer peer, bool includeDetails = false)
         {
-            if (peer.Node is null)
+            PeerValidator.ValidatePeer(peer);
+
+            var capabilities = ExtractCapabilities(peer);
+
+            SetBasicInfo(peer, capabilities);
+            SetNetworkInfo(peer);
+            SetProtocols(capabilities);
+
+            if (includeDetails)
             {
-                throw new ArgumentException(
-                    $"{nameof(PeerInfo)} cannot be created for a {nameof(Peer)} with an unknown {peer.Node}");
+                SetDetailedFields(peer);
             }
+        }
 
+        private void SetBasicInfo(Peer peer, IReadOnlyList<Capability> capabilities)
+        {
+            Id = peer.Node.Id;
             Name = peer.Node.ClientId;
-            Id = peer.Node.Id.Hash.ToString(false);
-            Host = peer.Node.Host is null ? null : IPAddress.Parse(peer.Node.Host).MapToIPv4().ToString();
-            Port = peer.Node.Port;
-            Address = peer.Node.Address.ToString();
-            IsBootnode = peer.Node.IsBootnode;
-            IsStatic = peer.Node.IsStatic;
             Enode = peer.Node.ToString(Node.Format.ENode);
-            Inbound = peer.InSession is not null;
+            Caps = capabilities;
+            Enr = EnrExtractor.GetEnrFromPeer(peer);
```

Comment:
Why complicate with extra class and return null. Just set null with the proper comment here. 

Btw I doubt it should be always null.

Reply:
Talk with @flcl42 about ENR

Reply:
you haven't applied this comment, please read one more time